src/util/constructor_of.h

```diff
@@ -0,0 +1,33 @@
+/*******************************************************************\
+
+Module: constructor_of
+
+Author: Diffblue Ltd.
+
+\*******************************************************************/
+
+#ifndef CPROVER_UTIL_CONSTRUCTOR_OF_H
+#define CPROVER_UTIL_CONSTRUCTOR_OF_H
+
+/// A type of functor which wraps around the set of constructors of a type.
+/// \tparam constructedt: The type which this functor constructs.
+template <typename constructedt>
```

Comment:
:question: What is the use of this? Currently only used in the unit test so was thinking it should live there, but of course if it has general utility then here is fine

Reply:
Is used in a later commit.